pytorchfi/errormodels.py

```diff
@@ -0,0 +1,358 @@
+"""
+pytorchfi.errormodels provides different error models out-of-the-box for use.
+"""
+
+import random
+import torch
+from pytorchfi import core
+
+
+# ###################
+#  helper functions #
+# ###################
+def random_neuron_location(pfi_model):
+    conv = random.randint(0, pfi_model.get_total_conv() - 1)
+    c = random.randint(0, pfi_model.get_fmaps_num(conv) - 1)
+    h = random.randint(0, pfi_model.get_fmaps_H(conv) - 1)
+    w = random.randint(0, pfi_model.get_fmaps_W(conv) - 1)
+
+    return (conv, c, h, w)
+
+
+def random_batch_element(pfi_model):
+    return random.randint(0, pfi_model.get_total_batches() - 1)
+
+
+def random_neuron_location_conv(pfi_model, conv):
+    c = random.randint(0, pfi_model.get_fmaps_num(conv) - 1)
+    h = random.randint(0, pfi_model.get_fmaps_H(conv) - 1)
+    w = random.randint(0, pfi_model.get_fmaps_W(conv) - 1)
+
+    return (c, h, w)
+
+
+def random_weight_location(pfi_model):
+    loc = list()
+
+    corrupt_layer = random.randint(0, pfi_model.get_total_conv() - 1)
+    loc.append(corrupt_layer)
+
+    curr_layer = 0
+    for name, param in pfi_model.get_original_model().named_parameters():
+        if "features" in name and "weight" in name:
+            if curr_layer == corrupt_layer:
+                for dim in param.size():
+                    loc.append(random.randint(0, dim - 1))
+            curr_layer += 1
+
+    assert curr_layer == pfi_model.get_total_conv()
+    assert len(loc) == 5
+
+    return tuple(loc)
+
+
+def random_weight_location_conv(pfi_model, conv):
+    loc = list()
+
+    corrupt_layer = conv
+    loc.append(corrupt_layer)
+
+    curr_layer = 0
+    for name, param in pfi_model.get_original_model().named_parameters():
+        if "features" in name and "weight" in name:
+            if curr_layer == corrupt_layer:
+                for dim in param.size():
+                    loc.append(random.randint(0, dim - 1))
+            curr_layer += 1
+
+    assert curr_layer == pfi_model.get_total_conv()
+    assert len(loc) == 5
+
+    return tuple(loc)
+
+
+def random_value(min_val=-1, max_val=1):
+    return random.uniform(min_val, max_val)
+
+
+# #################################
+#    Neuron Perturbation Models   #
+# #################################
+# single random neuron error in single batch element
```

Comment:
For section headers:
```
"""
Neuron Perturbation Models
"""
```

We can remove function descriptions and put them in the documentation site

Reply:
fixed.